src/celeritas/setup/Problem.cc

```diff
@@ -718,6 +718,22 @@ ProblemLoaded problem(inp::Problem const& p, ImportData const& imported)
                         = std::make_shared<optical::Transporter>(
                             std::move(inp));
                 },
+                [&](inp::OpticalTrackOffload) {
+                    // Build optical track transporter
+                    optical::Transporter::Input inp;
+                    inp.params = optical_params;
+                    if (action_times)
+                    {
+                        inp.action_times = ActionTimes::make_and_insert(
+                            optical_params->action_reg(),
+                            core_params->aux_reg(),
+                            "optical-action-times");
+                    }
+
+                    result.optical_transporter
+                        = std::make_shared<optical::Transporter>(
+                            std::move(inp));
```

Comment:
Consolidate this duplicate code with the above into a free function `build_optical_transporter` and use it for both cases:

```suggestion
                    result.optical_transporter
                        = build_optical_transporter(core_params, optical_params, action_times);
```